src/helpers.ts

```diff
@@ -254,3 +257,82 @@ export function getEncodedData(commands: string[], state: string[]): string {
   ]);
   return weirollWalletInterface.encodeFunctionData('executeWeiroll', [commands, state]);
 }
+
+export function getVerificationHash(script: WeirollScript, receiptToken: AddressArg, inputTokens: AddressArg[]) {
+  // TODO: confirm token order for encoding hash
+  return keccak256(
+    defaultAbiCoder.encode(
+      ['address[]', 'address', 'tuple(bytes32[], bytes[])'],
+      [inputTokens, receiptToken, [script.commands, script.state]],
+    ),
+  );
+}
+
+export async function getCampaign(
+  provider: StaticJsonRpcProvider,
+  chainId: number,
+  marketHash: string,
+): Promise<Campaign> {
+  const depositExecutorInterface = new Interface([
+    'function sourceMarketHashToDepositCampaign(bytes32 marketHash) external view returns (address owner, bool verified, uint8 numInputTokens, address receiptToken, uint256 unlockTimestamp, tuple(bytes32[] commands, bytes[] state) depositRecipe)',
+  ]);
+  const roles = getSimulationRolesByChainId(chainId);
+  const data = await provider.call({
+    to: roles.depositExecutor.address,
+    data: depositExecutorInterface.encodeFunctionData('sourceMarketHashToDepositCampaign', [marketHash]),
+  });
+  return depositExecutorInterface.decodeFunctionResult(
+    'sourceMarketHashToDepositCampaign',
+    data,
+  ) as unknown as Campaign;
+}
+
+export async function getWeirollWallets(
+  provider: StaticJsonRpcProvider,
+  chainId: number,
+  marketHash: string,
+): Promise<AddressArg[]> {
+  const depositExecutorInterface = new Interface([
+    'function getWeirollWalletByCcdmNonce(bytes32 marketHash, uint256 ccdmNonce) external view returns (address wallet)',
+  ]);
+  const roles = getSimulationRolesByChainId(chainId);
+
+  const wallets: AddressArg[] = [];
```

Comment:
Looking at this code and interacting with the DepositExecutor contract I assume that `ccdmNonce` starts at 1 and autoincrements monotonically.

Instead of "brute force" it to obtain the last weiroll wallet couldn't just filter the last [CCDMBridgeProcessed](https://80000.testnet.routescan.io/tx/0x6fba4bb63d183205773dfe864f831734fa11d7b0b26fb60243cc75289e4e2082/eventlog) event for the given sourceMarketHash (topic 1), and get the weirollWallet? Tbh, I have no idea how big this high this nonce can go, nor the performance of filtering backwards blocks.